### PR TITLE
Add port_usage tlv

### PR DESCRIPTION
**Pull request by shudongz**
Reviewer: @nick-bsn @andi-bigswitch 

Add tlv to indicate port usage.

### CODE REVIEW

**Comment by bsn-abat** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.bigswitch.com/job/loxigen_pull_req_docker/77/

**Comment by bsn-abat** [comment]
Find artifact changes from this pull request at https://github.com/floodlight/loxigen-artifacts/tree/pr-524-jenkins-loxigen_pull_req_artifacts-71

**Comment by bsn-abat** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.bigswitch.com/job/loxigen_pull_req_artifacts/71/

File: openflow_input/bsn_tlv [hunk 1]
```diff
@@ -963,3 +963,16 @@ struct of_bsn_tlv_data_mask : of_bsn_tlv {
     uint16_t length;
     of_octets_t value;
 };
+
+enum ofp_bsn_port_usage(wire_type=uint16_t) {
```
Comment (andi-bigswitch) [inline openflow_input/bsn_tlv]: Optional comment: 
Could alternatively model this as a bitmask enum (see https://github.com/floodlight/loxigen/blob/master/openflow_input/standard-1.5#L1241, with transmit/receive being a bit).

Would result in a slightly cleaner way to ask the question "is receive enabled"?

**Comment by andi-bigswitch** [comment]
Besides optional comment, LGTM

File: openflow_input/bsn_tlv (hunk 1 above)
Comment (3mp4y5) [inline openflow_input/bsn_tlv]: Is this a OF config message, requesting the interface to only be used in a specific direction?

Comment (shudongz) [inline openflow_input/bsn_tlv]: It's advisory to switchlight when allocating tx/rx buffers to physical interfaces. Larger buffers are more resilient against packet loss due to micorbursts.

**Comment by nick-bsn** [comment]
LGTM

**Comment by shudongz** [comment]
SELF-ACK

**Comment by bsn-abat** [comment]
ABAT: START_MERGE:  Log file at http://vss1/abat/2016.10.27.2151-m.loxigen.master/abat.log

**Comment by bsn-abat** [comment]
ABAT: ACCEPT:  Successfully merged

In case you want to see the build log, check out :
Log file at http://vss1/abat/2016.10.27.2151-m.loxigen.master/abat.log
/cc

**Pull request merged**